### ISSUE TITLE
False positive redundant return detected

### ISSUE THREAD

**Issue by luisdavim** [bbatsov/rubocop#4386]
Hi, some methods, specially the ones ending with a `!` don't always return what you want you method to return and you do need to use `return`, see the following example:
```
      def slice_host(slice, prefix)
        prefix.chars.zip(slice.chars).each do |char1, char2|
          break if char1 != char2
          slice.slice!(char1)
        end
        slice.slice!('.') if slice.chars.first == '.'
        return slice
      end
```
In the function above, rubocop complains about redundant return detected, however, if I don't use `return slice` the function won't return what I expect it to return.

**Comment by backus** [comment]
It just wants you to replace `return slice` with `slice` on the last line

**Comment by luisdavim** [comment]
And what is the problem of using `return`?

**Comment by deivid-rodriguez** [comment]
@luisdavim It's not a problem, it's just a matter of style and consistency. If you don't like the rule, you can disable it in your rubocop config:

```
Style/RedundantReturn:
  Enabled: false
```

PS: I think it would make sense to add the ability for this cop to enforce _always_ using explicit returns.

**Comment by Drenmi** [comment]
@deivid-rodriguez: There's a [feature request](https://github.com/bbatsov/rubocop/issues/4149) for that. 🙂
